gcs/get_gcs.go

```diff
@@ -9,16 +9,16 @@ import (
 	"strings"
 
 	"cloud.google.com/go/storage"
+	"github.com/hashicorp/go-getter/v2"
 	"google.golang.org/api/iterator"
 )
 
-// GCSGetter is a Getter implementation that will download a module from
+// Getter is a Getter implementation that will download a module from
 // a GCS bucket.
-type GCSGetter struct {
-	getter
```

Comment:
Why did you remove this? 🤔
I'm asking because the SmbGetter is using this to check if there's an available FileGetter. We need to check that In case the user doesn't want to provide the FileGetter and then SmbGetter can't use it.

Reply:
Ah, good question, I removed it because it is unused and is a bad pattern.

It was initially added in the master branch where the client/request are the same object. It would have allowed to reuse the getters of a request to do things like downloading the checksum from a file. It turns out it was not used.

I think this pattern was pretty bad and was going away from the initial design of this library which was aiming at avoiding a getter from knowing about the client ( so separating the concerns ).

Now if the `SmbGetter` needs this; let's pause a minute to think a little more; I need to re-read your PR with that in mind.